### ISSUE TITLE
Bad redirect after login

### ISSUE THREAD

**Issue by Rpd-Strike** [TeamUnibuc/SmartForms#39]
Although the current login procedure works fine from the url `smartforms.ml`
It doesn't work properly when developing from localhost, or from whatever link that doesn't usee port `80`

How to fix:
The route `/api/user/auth/` should redirect to the homepage, but changing to the correct port.
A hackish way, but somewhat reliable is to scan the `LOGIN_REDIRECT_URL` for the string `:[SERVER_PORT]` and change it to `:3000` if the redirect url also contains `localhost`.
Basically, change whatever the port is `LOGIN_REDIRECT_URL` configured with to 3000 if it is a localhost url.

Where to modify: [user_router.py:76](https://github.com/TeamUnibuc/SmartForms/blob/3731cc041f2f6fae5fad320b3aaeed4ba53b43bb/backend/sources/routers/user_router.py#L76)

**Comment by Rpd-Strike** [comment]
This happens because after the first redirect to google, fastAPI believes the request came from localhost:5000 instead of localhost:3000.
This is the reason the logout button works

Of course, other solutions may exist, maybe fastAPI has a "soft redirect" or smthg idk

**Comment by theodormoroianu** [comment]
You can specify where oauth should redirect us, via [this env variable](https://github.com/TeamUnibuc/SmartForms/blob/main/backend/.env.sample#L22).

If setting it doesn't work, it's probably because Google has a whitelist of allowed URIs, and we should add the one you are using to its list.

Does changing the env variable solve your issue? @Rpd-Strike

**Comment by Rpd-Strike** [comment]
I can test this if you add the url `http://localhost:3000/`

However I don't think it will work as intended because of the way it is designed right now:

```python
@router.get('/login')
async def login(request: Request):
    # Redirect Google OAuth back to our application
    
    redirect_uri = os.getenv("LOGIN_REDIRECT_URL")
    # request.url_for('auth')
    return await oauth.google.authorize_redirect(request, redirect_uri)

@router.get('/auth')
async def auth(request: Request):
    # Perform Google OAuth
    token = await oauth.google.authorize_access_token(request)

    # handle multiple behaviours the token has:
    # sometimes it gives back a dict with the "userinfo" object
    # other times we have to parse it from the id_token
    # most probably due to various versions of the oauth lib.
    if "userinfo" in token:
        user = token['userinfo']
    else:
        user = await oauth.google.parse_id_token(request, token)
    
    logger.info(f"User logged in: {user['email']}")
    
    # Save the user
    request.session['user'] = dict(user)

    return RedirectResponse(url='/')
```

So, after login im redirected to auth so that the user data can be saved properly and also call that function parse_id_token if necessary. If i were to redirect back to the frontend then the user data won't be correctly saved. Maybe we should have 2 redireect URIs in env? one for redirecting to the `auth` from `login` that uses google oauth and another one to redirect us to the frontend from `auth`

**Comment by Rpd-Strike** [comment]
How we did it previously is we have 2 environment variables, one for the frontend one for the backend, and then in the code you just concatenate the url with the route you need, and like that you dont have to worry to put the correct route in the `env` file, the actual code is. Seeems like a good separation of responsibilities to me. WDYT

**Comment by Rpd-Strike** [comment]
I made a  PR that should address this: #43 
If you want to take a look @theodormoroianu

**Comment by theodormoroianu** [comment]
I updated the PR a little bit:

I removed references to the `BACKEND_URL`, as it's already accessible via the proxy made in `FRONTEND_URL/api`.
This way, we only have to worry about the `FRONTEND_URL`.
